cmake/dependencies/openPMD.cmake

```
@@ -87,7 +87,7 @@ if(WarpX_OPENPMD)
     set(WarpX_openpmd_repo "https://github.com/openPMD/openPMD-api.git"
         CACHE STRING
         "Repository URI to pull and build openPMD-api from if(WarpX_openpmd_internal)")
-    set(WarpX_openpmd_branch "0.16.1"
+    set(WarpX_openpmd_branch "c734fb50af4ebb4ea3c3a9c14075bb10372e68df"
```

trying https://github.com/openPMD/openPMD-api/pull/1824
```suggestion
    set(WarpX_openpmd_branch "fix-gcc12-false-positives"
```